### PR TITLE
refactor: CLI creator class

### LINKED ISSUE #1877: refactor: Post group-design cleanup
**Describe the bug**
We've migrated from group design, all that's left is to cleanup `GroupHelper` and unnecessary computation in `webpack-cli.js` eg we don't need to compute group map etc. 

This issue is to list down potential cleanups which can be done

**What is the current behavior?**
Extra methods in class doing unnecessary computation

**To Reproduce**

**Expected behavior**
Only keeping relevant methods.

**Screenshots**

**Please paste the results of `webpack-cli info` here, and mention other relevant information**

**Additional context**
I have some ideas, will put up a PR for feedback.

@evilebottnawi also had some ideas on how to go about it.

### PR DESCRIPTION
**Pull request by anshumanv**
**What kind of change does this PR introduce?**

refactor

**Did you add tests for your changes?**
Yes

**If relevant, did you update the documentation?**

NA
**Summary**

- Cleanup webpack-cli.js redundant class methods
- Remove stale group helpers

**Does this PR introduce a breaking change?**

No

**Other information**
Fixes #1877

### CODE REVIEW

**Review by evilebottnawi — APPROVED** [review]
/cc @webpack/cli-team

**Review by snitin315 — APPROVED** [review]

File: packages/webpack-cli/lib/webpack-cli.js [hunk 1]
```diff
@@ -19,48 +17,33 @@ const resolveStats = require('./groups/resolveStats');
...
-        }
+        const coreConfig = Object.keys(parsedArgs)
+            .filter((arg) => {
+                return coreFlagMap.has(toKebabCase(arg));
+            })
+            .reduce((acc, cur) => {
+                acc[toKebabCase(cur)] = parsedArgs[cur];
+                return acc;
+            }, {});
+        const coreCliHelper = require('webpack').cli;
```
Comment (evilebottnawi) [inline packages/webpack-cli/lib/webpack-cli.js:42]: What about if we have webpack@4? We don't have `.cli`

Comment (evilebottnawi) [reply packages/webpack-cli/lib/webpack-cli.js:42]: /cc @anshumanv

Comment (anshumanv) [reply packages/webpack-cli/lib/webpack-cli.js:42]: oops, I think it worked, lemme recheck

Comment (evilebottnawi) [reply packages/webpack-cli/lib/webpack-cli.js:42]: Yep, let's check it, but ci success so I think it should be work

Comment (anshumanv) [reply packages/webpack-cli/lib/webpack-cli.js:42]: strange, passed in CI but fails locally 😞

Comment (anshumanv) [reply packages/webpack-cli/lib/webpack-cli.js:42]: sending a follow up - sorry

**Pull request merged**